tokio/src/util/linked_list.rs

```diff
@@ -66,27 +70,40 @@ unsafe impl<T: Sync> Sync for Pointers<T> {}
 
 // ===== impl LinkedList =====
 
-impl<T: Link> LinkedList<T> {
+impl<L, T> LinkedList<L, T> {
+    /// Constant constructor for a `LinkedList`.
+    #[allow(dead_code)] // NOTE: This will get removed with: https://github.com/tokio-rs/tokio/pull/2790
+    pub(crate) const fn const_new() -> LinkedList<L, T> {
+        LinkedList {
+            head: None,
+            tail: None,
+            _marker: PhantomData,
+        }
+    }
+}
+
+impl<L: Link> LinkedList<L, L::Target> {
     /// Creates an empty linked list
-    pub(crate) fn new() -> LinkedList<T> {
+    pub(crate) fn new() -> LinkedList<L, L::Target> {
         LinkedList {
             head: None,
             tail: None,
+            _marker: PhantomData,
         }
     }
```

Comment:
I think it would be fine to just have one `const fn new` constructor for this type.

My suggestion that we should have separate `new` and `const_new` constructors was specifically for the public API types, where the constructor can only be a `const fn` when the `"parking_lot"` feature flag is enabled --- in that case, i thought we would prefer for the feature flag to add a separate function rather than changing the signature of an existing one.